### DISCUSSION TITLE
Affichage Trace relatif

### DISCUSSION THREAD

**Discussion by ArmandGC** [witsa/synapps#590] [Q&A] [answered]
Bonjour,
Je suis avec sur un Bâtiment avec 60 logements et 3 compteurs par logement, j'affiche actuellement un composite de mes 3 compteurs avec un chemin relatif pour chaque logement, est-il possible de faire de même pour l'affichage de trace ?
de pouvoir dans mon composite ajouter un bouton qui m'ouvre une modale avec la trace de la puissance par exemple et qu'il prenne en compte mon chemin relatif par bâtiment.
j'ai essayer de faire avec un fournisseur de ressource et de faire un lien /fournisseur de variable avec mon fournisseur de ressource en clé parent mais le graph ne semble pas le prendre en charge.

**Comment by WITNicolas** [comment, marked as answer]
Bonjour,

Si j'ai bien compris, il suffit de mettre dans l'acteur grapheur, le chemin de la ressource concernée (en supposant que la trace et dans la ressource évidement).
Dans la laison Traces/Flux, mettre donc le chemin désirée puis faire un transformation OnReadTransform
![image](https://github.com/witsa/synapps/assets/33632664/0640647e-92ff-4219-8b1e-b5be63b6519c)
![image](https://github.com/witsa/synapps/assets/33632664/7b9d3fa2-20c8-43c4-aee4-df1fd769ad55)


En ajoutant le Binding suivant :
return "TRA_"+context.value+".FNCT.F0001";

context.value est donc le chemin de la ressource, ce qui provient de la liaison
FNCT.F0001 la fonction Traces dans la ressource

L'acteur Grapheur va par conséquent concaténer et interpréter
TRA_:EASY.RESS.R0001.R0002.FNCT.F0001

Pour un résultat avec l'echelle automatique (dans l'exmple j'ai 2 acteurs Grapheurs sur ma modale)
![image](https://github.com/witsa/synapps/assets/33632664/5b807aba-7803-4a1a-aa6a-cba5c0b839bc)

**Reply by ArmandGC** [reply]
Super ça fonctionne bien effectivement,
Merci.
Par contre je viens de voir que les unités ne s'affichent pas, même si je met une préparation en plus (la préparation ne semble pas être prise en compte avec la trace). Est-il possible donc de rajouté des paramètres en plus après le lien vers la trace pour précisé une unité / une échelle min/max ... ?

**Reply by WITNicolas** [reply]
Il s'agit effectivement de la trace brute, la preparation graphique n'est pas prise en compte.
S'il faut utiliser les préparations graphiques, il faut passer des arguments à l'acteur GRAPHEUR pour qu'il utilise la propriété spécifique PREP_

Donc au niveau du composite, il faut rajouter une additionnelle texte 'propPREPA' (par exemple) dans laquelle on pourra saisir PREP_xxxxxx
Il s'agit du label de la préparation graphique.
Il faudra liaisonner la propriété spécifique Preparation de l'acteur Grapheur vers l'additionnelle 'propPREPA' du Composite.

Sur vos scenes, sur chaque instance du composite, vous pourrez compléter avec l'argument 'propPREPA' (comme pour les chemins des ressources en fait).
